### ISSUE TITLE
Create Installers for MacOS and Windows

### ISSUE THREAD

**Issue by gdams** [AdoptOpenJDK/openjdk-build#110]
We should create start creating installers for our binaries. The installation process should be pretty simple but we need to consider a few things:
- how to handle updates
- where the binary will be stored on the system
- will the installer include the binary or download it (web based on offline)
- do we need to codesign our installers?
- what language to write the installer in?

The way I view the installer working is as follows:
1. the user clicks `install` and the tarball is downloaded to the temp folder.
2. the installer checks the checksum to confirm that the tarball is correct.
3. the binary is then extracted to an AdoptOpenJDK folder that exists somewhere on the system.
4. the installer then removes all versions of AdoptOpenJDK from the `PATH` and updates with the new version. (the `JAVA_HOME` variable is also updated at this point)

There are a few ways to go about making this installer, I have already looked at electron as a possibility but have got stuck when it comes to permanently storing the AdoptOpenJDK path in the `PATH` environment variable.

Any help/suggestions welcome!

**Comment by sunnythepooh** [comment]
Installer tends to include the binary instead of downloading from Internet - what if you have a machine that you wanted to deploy the software but can't access the internet?

For the Windows Installer if you wanted to install into c:\program files it will required escalated access - otherwise it can only be installed in the user directory. Which do we prefer?

For Windows I would recommend using WIX toolkit (http://wixtoolset.org/). The tool is open sourced and it would generate MSI installer that can be used with most Windows management tools.

**Comment by gdams** [comment]
good point @sunnythepooh. Okay so I would say that this should go into `c:\program files`. Have you had much experience with wix toolkit? it would be great if you could put together a prototype installer for us to all take a look at

**Comment by gdams** [comment]
@sunnythepooh any progress with this?

**Comment by gouessej** [comment]
Hi

I had to write installers for my first person shooter, I created some executables with NSIS for Windows, DEB and RPM for GNU Linux and an APP bundle for OS X. Feel free to look at my source code:
https://sourceforge.net/p/tuer/code/HEAD/tree/pre_beta/src/main/xml/jndt/antlib.xml

**Comment by gdams** [comment]
Hi @gouessej, thanks for the links.... Is there any chance that you could put together a basic sample installer for our OpenJDK binaries?

**Comment by gouessej** [comment]
@gdams I can try to take a typical Windows 64-bit binary build in input and give you an executable installer based on NSIS when I have enough spare time. Would it help?

**Comment by gdams** [comment]
@gouessej I know that @sunnythepooh started work on the windows installer but I'm not sure how far he got, it would be great if you could take a look at that for us!

**Comment by gouessej** [comment]
@gdams The problem is that we use 2 distinct approaches, he tried to create typical native Microsoft installers by using Wix whereas I found Wix too much complicated and preferred using NSIS. Both have their pros and cons. As I spend most of my time at home under GNU Linux, NSIS fits better into my needs as I can run it under GNU Linux. If you really want to stick with MSI, I won't be able to help you. If I had to run Wix under GNU Linux, I would need to use Wine and Wix + Wine are known to work only on simple packages and/or only not with the latest version (<3.5?).

**Comment by gdams** [comment]
closing as initial installers are now complete

**Comment by karianna** [comment]
@gdams Any ideas?
